### ISSUE TITLE
Cakebox booting slow when host not connected in the Internet

### ISSUE THREAD

**Issue by Havarem** [alt3/cakebox#57]
Don't know if it is desired behavior but when I start the VM and I'm not connected to the Internet, I've seen like a dozen or more of these `default: Warning: Connection timeout. Retrying...`. At some point it reaches the "maximum number of tries" but Cakebox always ends up running.

Is it possible for the box to be configured to try to connect to the Internet once the boot is completed, so even if Internet is not available I'll be able to develop my apps quickly? We could add a feature in cakebox-console to "Try to connect to the Internet", something like that.

**Comment by bravo-kernel** [comment]
Thanks for reporting this, good to hear the box works on LAN-less as well (as designed). I will try to reproduce the behavior and see if we can improve the speed at points.

**Can you let us know** if this is happening with ``vagrant up`` or ``vagrant --reprovision``?

FYI:

+  the vagrant file already has some (of the more well known) fixes for slow booting in LAN-less environments, see https://github.com/alt3/cakebox/blob/dev/.cakebox/Vagrantfile.rb#L57 and L58.
+ we will probably add some inet-connectivity checks during provisioning as well

**Comment by Havarem** [comment]
I use `vagrant up`. I'm on MacOS X 10.10.3, with vagrant 1.7.2 and virtualbox 4.2.26. The Cakebox version was v1.0-alpha3. I, obviously, didn't have an Internet access when I experienced the problem.

I will look further into those settings. I'm brand new in vagrant (started to use two weeks ago) and I discover Cakebox by mere coincidence (I think from twitter last week or so).

Great job!

**Comment by bravo-kernel** [comment]
Closing this ticket as I have just tested booting in a completely LAN-less environment and now believe that what you are seeing is not the result of not being connected to the internet but instead of default Vagrant 1.7.x behavior determining which SSH key to use for connecting to the box and/or inserting a secure key (given the ``Connection timeout`` errors you describe).

This makes perfect sense as there are no additional steps in the ``vagrant up`` process related to an active internet connection except for the DNS configuration inside the vm itself and for that.... the two fixes mentioned earlier are already in place.

Long story short.... once the Vagrant SSH key detection/insertion process has completed the booting process should continue at normal speed, regardless of inet-connectivity.
